internal/scheduler/schedulerobjects/nodetype.go

```diff
@@ -67,40 +66,44 @@ func NewNodeType(taints []v1.Taint, labels map[string]string, indexedTaints map[
 	}
 }
 
-// nodeTypeIdFromTaintsAndLabels generates an id that is unique for each combination
-// of taints, labels, and unset labels, of the form
-// $taint1$taint2...&$label1=labelValue1$label2=labelValue2...&$unsetIndexedLabel1=unsetIndexedLabelValue1...
+// nodeTypeIdFromTaintsAndLabels generates a id unique for each combination of taints, labels, and unset labels.
+// The id is based on the fnv1a hash. Hash collisions do not affect correctness, only the efficiency of sorting out nodes.
 //
 // We separate taints/labels by $, labels and values by =, and and groups by &,
 // since these characters are not allowed in taints and labels; see
 // https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
 // https://man.archlinux.org/man/community/kubectl/kubectl-taint.1.en
-func nodeTypeIdFromTaintsAndLabels(taints []v1.Taint, labels, unsetIndexedLabels map[string]string) string {
-	// TODO: To reduce key size (and thus improve performance), we could hash the string.
+func nodeTypeIdFromTaintsAndLabels(taints []v1.Taint, labels, unsetIndexedLabels map[string]string) uint64 {
```

Comment:
I've changed this from a string to a uint64. As a result, it's of fixed-length. While unlikely, I believe the previous varying-length id could cause issues with the compound key used for node indexing. As a bonus, using a uint64 also improves performance.